modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/Editor.es.js

```diff
@@ -437,6 +452,29 @@ export function Editor({
 		init
 	);
 
+	const InputOutputLength = ({target, url}) => {
+		const {resource} = useResource({
+			fetch,
+			link: location.origin + url,
+			variables: {
+				ddmDataProviderInstanceId: target,
+			},
+		});
+
+		return resource?.inputs?.length + resource?.outputs?.length;
```

Comment:
Can `resource` , `resource.inputs` and `resource.outputs` be really `undefined` or `null`? How gonna the sum behaves in this case?
```javascript
1 + 2 // 3
null + null // 0
undefined + undefined // NaN
```

Reply:
If NaN or Null the validation will be true, since there's no options to be selected. Those cases must be rare, since it doesn't make sense to me to create a data provider without outputs, at least.